Realm/Realm/Configurations/SyncConfiguration.cs

```diff
@@ -34,6 +33,10 @@ namespace Realms.Sync
     /// <seealso href="https://docs.mongodb.com/realm/sync/overview/">Sync Overview Docs</seealso>
     public class SyncConfiguration : RealmConfigurationBase
     {
+        // TODO: https://github.com/realm/realm-dotnet/issues/2571 - we should update the public property type to be RealmValue and
+        // get rid of this field.
+        internal readonly RealmValue _partition;
```

Comment:
We were storing Partition as an object, making it harder to do comparisons between partitions of different types. This addresses https://github.com/realm/realm-dotnet/security/code-scanning/621?query=ref%3Arefs%2Fheads%2Fmaster